projects/RabbitMQ.Client/client/impl/RpcContinuationQueue.cs

```diff
@@ -122,7 +123,7 @@ public IRpcContinuation Next()
         /// waiting continuations.
         ///</para>
         ///</remarks>
-        public bool TryPeek<T>(out T continuation) where T : IRpcContinuation
+        public bool TryPeek<T>([MaybeNullWhen(false)] out T continuation) where T : IRpcContinuation
```

Comment:
```suggestion
        public bool TryPeek<T>([NotNullWhen(true)] out T? continuation) where T : IRpcContinuation
```

Reply:
what would be the benefit of this? Except making it worse if T would be a struct.

Reply:
`T` technically can be a struct, but the only usage is with T being ChannelCloseAsyncRpcContinuation class. My suggestion would make sense only if T is restricted to class.
The befit was clarity of contract.
As T can be any type, there is nothing to do for this comment. Resolved.

Reply:
I've limited TryPeek to `class` and accepted @lechu445's suggestion.

Reply:
This is the same thing as [`public bool TryGetValue(TAlternateKey key, [MaybeNullWhen(false)] out TValue value)`](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Queue.cs,250).

Further more, if `T` is not supposed to be a `struct`, it can be constrained to be a `class`.

Reply:
This is why I mostly use `[MaybeNullWhen(false)]` because it's always correct / performant irrespective what type it is.

syntactially AFAIK those are equal (`[MaybeNullWhen(false)]` and `[NotNullWhen(true)]`)

Reply:
`[MaybeNullWhen(false)]` indicates that the value may be `null` if the method returns `false`, but might not be `null`.

`[NotNullWhen(true)]` indicates that the value is definitely not `null` when the method returns `true`.

All this based on the annotations and developers honoring them.